awscrt/common.py

```diff
@@ -18,3 +18,23 @@ def get_cpu_count_for_group(group_idx: int) -> int:
     Returns number of processors in a given group.
     """
     return _awscrt.get_cpu_count_for_group(group_idx)
+
+class SystemEnvironment:
+
+    def __init__(self):
+        self._env = _awscrt.load_system_environment()
+
+        if self.is_ec2_nitro_instance():
+            self._detected_instance_type = _awscrt.get_ec2_instance_type(self._env)
+        else:
+            self._detected_instance_type = None       
+
+    def is_ec2_nitro_instance(self) -> bool:    
+        return _awscrt.is_env_ec2(self._env)
+    
+    def get_ec2_instance_type(self) -> str:        
```

Comment:
the underylying C function is `get_virtualization_product_name()`

let's consistently call it "ec2" or "virtualization_product_name" but it's annoying to change terminology between the layers

Reply:
ec2 and vritualization product name are not the same thing. ec2 is A vendor, and aws-c-common is mostly generic, I did move most of this to aws-c-s3 however and it makes more sense there. I can clean it up here.